src/plugins/MamicoCoupling.h

```diff
@@ -2,114 +2,135 @@
- * */
-	void setMamicoMacroscopicCellService(coupling::services::MacroscopicCellService<3>* macroscopicCellService){
-		_macroscopicCellService = static_cast<coupling::services::MacroscopicCellServiceImpl<ls1::LS1RegionWrapper,3>*>
-			(macroscopicCellService);
-	}
+  /**
+   * Sets the macroscopicCellService object that controls the inner coupling
+   * logic.
+   *
+   * MaMiCo extracts the MamicoCoupling plugin object from the simulation object
+   * after initialization and uses this function to set the
+   * macroscopicCellCervice. The code for this object can be found in
+   * https://github.com/HSU-HPC/MaMiCo/blob/master/coupling/services/MacroscopicCellService.cpph
+   */
+  void setMamicoMacroscopicCellService(
+      coupling::services::CouplingCellService<3> *couplingCellService) {
+    _couplingCellService =
+        static_cast<coupling::services::CouplingCellServiceImpl<
+            ls1::LS1RegionWrapper, 3> *>(couplingCellService);
+  }
 #endif
 
-/** @brief Enables coupling logic, allowing coupling steps to run while simulation is running.
- * 
- * Set from within MaMiCo, check https://github.com/HSU-HPC/MaMiCo/blob/master/coupling/interface/MDSimulationFactory.h, class LS1MDSimulation
- * */
-	void switchOnCoupling(){ _couplingEnabled = true; }
-
-/** @brief Disables coupling logic, not allowing coupling steps to run. Typically done when equilibrating the simulation initially.
- * 
- * Set from within MaMiCo, check https://github.com/HSU-HPC/MaMiCo/blob/master/coupling/interface/MDSimulationFactory.h, class LS1MDSimulation
- * */
-	void switchOffCoupling(){ _couplingEnabled = false; }
-
-/** @brief Getter method for the coupling state.
- * 
- * Not used currently, but may be used in future
- * */
-	bool getCouplingState() { return _couplingEnabled;}
-
+  /**
+   * Enables coupling logic, allowing coupling steps to run while simulation is
+   * running.
+   *
+   * Set from within MaMiCo, check
+   * https://github.com/HSU-HPC/MaMiCo/blob/master/coupling/interface/MDSimulationFactory.h,
+   * class LS1MDSimulation
+   */
+  void switchOnCoupling() { _couplingEnabled = true; }
+
+  /**
+   * Disables coupling logic, not allowing coupling steps to run. Typically done
+   * when equilibrating the simulation initially.
+   *
+   * Set from within MaMiCo, check
+   * https://github.com/HSU-HPC/MaMiCo/blob/master/coupling/interface/MDSimulationFactory.h,
+   * class LS1MDSimulation
+   */
+  void switchOffCoupling() { _couplingEnabled = false; }
+
+  /**
+   * Getter method for the coupling state.
+   *
+   * Not used currently, but may be used in future
+   */
+  bool getCouplingState() { return _couplingEnabled; }
 
 private:
 #ifdef MAMICO_COUPLING
-	coupling::services::MacroscopicCellServiceImpl<ls1::LS1RegionWrapper,3>* _macroscopicCellService;
+  coupling::services::CouplingCellServiceImpl<ls1::LS1RegionWrapper, 3>
+      *_couplingCellService;
```

Comment:
```suggestion
  coupling::services::CouplingCellServiceImpl<ls1::LS1RegionWrapper, 3>
      *_couplingCellService = nullptr;
```
Missing initialization